spec/src/main/asciidoc/property-expansion.asciidoc

```diff
@@ -0,0 +1,70 @@
+//
+// Copyright (c) 2020 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// Contributors:
+// Roberto Cortez
+
+[[property-expansion]]
+== Property Expansion
+
+The value of a configuration property may contain an expression corresponding to another configuration property. An
+expression string is a mix of plain strings and expression segments, which are wrapped by the sequence `${ ... }`.
+
+Consider the following configuration properties file:
+
+[source,properties]
+----
+server.url=http://${server.host}/endpoint
+server.host=example.org
+----
+
+When looking up the `server.url` property, the value will be resolved and expanded to `http://example.org/endpoint`.
+All MicroProfile Config rules still apply. The `Config` is able to resolve expressions from different ConfigSources.
+
+Additionally, it is also possible to use the following syntax with for property expansion:
+
+* `${expression:value}` - Provides a default value after the `:` if the expansion doesn't find a value.
+* `${my.prop${compose}}` - Composed expressions. Inner expressions are resolved first.
+* `${my.prop}${my.prop}` - Multiple expressions.
```

Comment:
What requirements on implementors are there regarding cycles?

Reply:
You mean recursion level in composed expressions?